middleware/proxy_1_11.go

```diff
@@ -3,21 +3,29 @@
 package middleware
 
 import (
+	"context"
 	"fmt"
 	"net/http"
 	"net/http/httputil"
 
 	"github.com/labstack/echo/v4"
 )
 
+// StatusCodeContextCanceled is HTTP status code for when client closed connection
+const StatusCodeContextCanceled = 499
```

Comment:
Status code 499 is non-standard HTTP status code that is used by Nginx as much I find by googling. 

If your application needs different codes for this case you could handle this case in echo error handler or wrap Proxy middleware into wrapper function that handles this case. 

`echo.HTTPError` has field `Internal` which you can check in your own error handling code

so this IF should be instead to provide this access
```go
		httpError := echo.NewHTTPError(http.StatusBadGateway, fmt.Sprintf("remote %s unreachable, could not forward: %v", desc, err))
		httpError.Internal = err
		c.Set("_error", httpError)
```

NB: there are no tests. please add.

Reply:
@aldas I see other reverse proxies like caddy that do this on context canceled requests. (https://github.com/caddyserver/caddy/pull/3816/commits/db715e3e496d5fd00e7fb607ac6664d161ae1dcb)
Isn't that your opinion to do this for context canceled requests?

Reply:
Could you add longer description to that constant - similarly like Caddy does (for people looking at it in the future and wondering why non-standard code) and set original error as internal field for both cases.  In that way we handle special case of context cancellation and provide way for other library users to have access to original error through `httpError.Internal` field

Reply:
Done

Reply:
Text suggestion
```
// StatusCodeContextCanceled is a custom HTTP status code for situations
// where a client unexpectedly closed the connection to the server.
// As there is no standard error code for "client closed connection", but
// various well-known HTTP clients and server implement this HTTP code we use
// 499 too instead of the more problematic 5xx, which does not allow to detect this situation
```